types/index.d.ts

```diff
@@ -4,8 +4,18 @@ export type config = {schema?: Schema, initialData?: Transaction };
 export type Entity = {
   get: (...params:string[]) => any;
 }
+export type Fact = [number, string, string | number | object | Array<any>, number, boolean];
+export type databaseClient = {
+  dbToString: () => string,
+  dbFromString: (dbString: string) => databaseClient,
+  dbToDatoms: () => Fact[],
+  addTransactListener: (Facts: Fact[]) => void,
```

Comment:
addTransactionListener

And this takes a listener function, not an array of datoms. That function takes datoms... IDK if that can be expressed in TS

Reply:
@smothers what does the listener function return?

Reply:
my bad, this is correctly named `addTransactListener`
the other one should be `removeTransactListener`

Reply:
and this returns and `any`. We should probably default to `any` instead of `void` for these typings given cljs always just returns the last expression. If there's a good reason to change that we can be more intentional and make the last expression null or undefined for effectful functions like these.

Reply:
yeah, just did some testing and all the voids should be changed to any given these functions return clojure datastructures and I don't think TS is ok calling those void https://www.typescriptlang.org/docs/handbook/basic-types.html#void.